nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -42,24 +49,26 @@ def system_dir(profile, generation):
 efi /efi/memtest86/BOOTX64.efi
 """
 
-def write_loader_conf(profile, generation):
+def generation_conf_filename(profile, generation, specialisation):
+    profile_part = f"-{profile}" if profile else ""
+    specialisation_part = f"-specialisation-{specialisation}" if specialisation else ""
```

Comment:
About style, maybe use ".".join([...]) instead?

Reply:
I kept the `-`, but did a join instead, which I think is a bit clearer.

Reply:
yea, right, the dot was a mistake at my side